src/TextInput/TextInput.tsx

```diff
@@ -0,0 +1,323 @@
+   */
+  autoCapitalize?: 'none' | 'sentences' | 'words' | 'characters'
+
+  /** Vertically align text when `multiline` is set to true */
+  textAlignVertical?: 'auto' | 'top' | 'bottom' | 'center'
+
+  /**
+   * If false, disables auto-correct.
+   * The default value is true.
+   */
+  autoCorrect?: boolean
+
+  /**
+   * If false, scrolling of the text view will be disabled. The default value is true. Only works with multiline={true}
+   */
+  scrollEnabled?: boolean
+
+  /**
+   * Limits the maximum number of characters that can be entered.
+   * Use this instead of implementing the logic in JS to avoid flicker.
+   */
+  maxLength?: number
+
+  /**
+   * enum("default", 'numeric', 'email-address', "ascii-capable", 'numbers-and-punctuation', 'url', 'number-pad', 'phone-pad', 'name-phone-pad',
+   * 'decimal-pad', 'twitter', 'web-search', 'visible-password')
+   */
+  keyboardType?: KeyboardTypeOptions
+
+  /** Display the error message at the bottom */
+  errorText?: string
+
+  /** Props to be passed to the error text component */
+  errorProps?: TextProps
+
+  /**
+   * If true, focuses the input on componentDidMount.
+   * The default value is false.
+   */
+  autoFocus?: boolean
+
+  /** Whether the input can have multiple lines */
+  multiline?: boolean
+
+  /** The string that will be rendered before text input has been entered */
+  placeholder?: string
+
+  /** The text color of the placeholder string */
+  placeholderTextColor?: ColorValue
+
+  /**
+   * enum('default', 'go', 'google', 'join', 'next', 'route', 'search', 'send', 'yahoo', 'done', 'emergency-call')
+   * Determines how the return key should look.
+   */
+  returnKeyType?: ReturnKeyTypeOptions
+
+  /**
+   * If true, the text input obscures the text entered so that sensitive text like passwords stay secure.
+   * The default value is false.
+   */
+  secureTextEntry?: boolean
+
+  /** Callback that is called when the text input's text changes. Changed text is passed as an argument to the callback handler */
+  onChangeText?: (text: string) => void
+
+  /** Callback that is called when the text input is focused */
+  onFocus?: () => void
+
+  /** Callback that is called when the text input's submit button is pressed */
+  onSubmitEditing?: () => void
+
+  /** Callback that is called when the text input is blurred */
+  onBlur?: (args: any) => void
+}
+
+interface CustomIconProps {
+  source: ImageSourcePropType
+  size?: number
+  color?: string
+  resizeMode?: ImageResizeMode
+  iconContainerStyle?: StyleProp<ViewStyle>
+  iconStyle?: StyleProp<ImageStyle>
+  style?: StyleProp<ViewStyle>
+}
+
+interface TextInputContainerProps {
+  multiline?: boolean
+}
+
+const CustomIcon: React.FC<CustomIconProps> = ({
+  source,
+  size,
+  color,
+  resizeMode,
+  iconContainerStyle,
+  iconStyle,
+}) => (
+  <IconWrapper style={iconContainerStyle}>
+    <Icon source={source} size={size} color={color} resizeMode={resizeMode} style={iconStyle} />
+  </IconWrapper>
+)
```

Comment:
can you write an Icon component used for the project?
The Button component can use the Icon component

Reply:
done